xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -217,19 +220,28 @@ float CEngineStats::GetCacheTotal()
 
 float CEngineStats::GetMaxDelay() const
 {
-  return static_cast<float>(MAX_CACHE_LEVEL) + static_cast<float>(MAX_WATER_LEVEL) +
-         m_sinkCacheTotal;
+  return MAX_CACHE_LEVEL + MAX_WATER_LEVEL + m_sinkCacheTotal;
 }
 
 float CEngineStats::GetWaterLevel()
 {
   std::unique_lock<CCriticalSection> lock(m_lock);
+
   if (m_pcmOutput)
+  {
     return static_cast<float>(m_bufferedSamples) / m_sinkSampleRate;
+  }
   else
-    return static_cast<float>(m_bufferedSamples *
-                              m_sinkFormat.m_streamInfo.GetDuration(m_sinkNeedIecPack)) /
-           1000;
+  {
+    float wl = static_cast<float>(m_bufferedSamples *
+                                  m_sinkFormat.m_streamInfo.GetDuration(m_sinkNeedIecPack)) /
+               1000;
+
+    if (wl >= MAX_WATER_LEVEL)
+      wl = MAX_WATER_LEVEL - 0.00001f;
```

Comment:
Don't you want to return wl - 0.000001f to compensate for rounding errors. In error case there might be more far more buffered samples in there and AE would not realize.

Reply:
```
if (wl > MAX_WATER_LEVEL && wl < MAX_WATER_LEVEL + EPSILON)
  wl = MAX_WATER_LEVEL;
```